hack/test-go.sh

```diff
@@ -135,16 +135,21 @@ if [[ $iterations -gt 1 ]]; then
 fi
 
 if [[ -n "${1-}" ]]; then
-  covdir="/tmp/k8s_coverage/$(kube::util::sortable_date)"
-  kube::log::status "Saving coverage output in '${covdir}'"
+  cover_params=""
+  if [[ -n "${KUBE_COVER}" ]]; then
+    covdir="/tmp/k8s_coverage/$(kube::util::sortable_date)"
+    kube::log::status "Saving coverage output in '${covdir}'"
+    mkdir -p "${covdir}/${arg}"
+    eval "cover_params=(${KUBE_COVER} -coverprofile=\"${covdir}/${arg}/coverage.out\")"
```

Comment:
Please use a bash array for cover_params instead.

```shell
  cover_params=()
  if [[ ... ]]; then
    ...
    cover_params=(${KUBE_COVER} -coverprofile="${covdir}/${arg}/coverage.out")
  fi
  ...
  go test ... "${cover_params[@]+${cover_params[@]}"
```

The last part (`${...+${...}}`) will ensure it will not trigger "nounset" when the array is empty.

Does that make sense?

Please test it with different combinations to make sure it works on all of them... I wrote this snippet from memory without any testing, I think it works but please make sure of it.

Cheers,
Filipe